atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -454,29 +454,26 @@ Set<CassandraServer> validateNewHostsTopologiesAndMaybeAddToPool(
         if (serversToAdd.isEmpty()) {
             return Set.of();
         }
-        Set<CassandraServer> serversToAddWithoutOrigin = serversToAdd.keySet();
-        Map<CassandraServer, CassandraClientPoolingContainer> serversToAddContainers =
-                getContainerForNewServers(serversToAddWithoutOrigin);
 
+        Set<CassandraServer> serversToAddWithoutOrigin = serversToAdd.keySet();
         Preconditions.checkArgument(
-                Sets.intersection(currentContainers.keySet(), serversToAddContainers.keySet())
+                Sets.intersection(currentContainers.keySet(), serversToAddWithoutOrigin)
                         .isEmpty(),
                 "The current pool of servers should not have any server(s) that are being added. This is unexpected"
                         + " and could lead to undefined behavior, as we should not be validating already validated"
                         + " servers. This suggests a bug in the calling method.",
-                SafeArg.of("serversToAdd", CassandraLogHelper.collectionOfHosts(serversToAddContainers.keySet())),
+                SafeArg.of("serversToAdd", CassandraLogHelper.collectionOfHosts(serversToAddWithoutOrigin)),
                 SafeArg.of("currentServers", CassandraLogHelper.collectionOfHosts(currentContainers.keySet())));
```

Comment:
We don't need to convert it to serversToAddContainers to have this check

Reply:
👍